include/ebpf_api.h

```diff
@@ -348,6 +354,114 @@ extern "C"
+     * @retval EBPF_INVALID_ARGUMENT One or more parameters are incorrect.
+     * @retval EBPF_NO_MEMORY Out of memory.
+     * @retval EBPF_ELF_PARSING_FAILED Failure in parsing ELF file.
+     * @retval EBPF_FAILED Some other error occured.
+     */
+    _Success_(result == EBPF_SUCCESS) ebpf_result_t ebpf_program_load(
+        _In_z_ const char* file_name,
+        _In_opt_ const ebpf_program_type_t* program_type,
+        _In_opt_ const ebpf_attach_type_t* attach_type,
+        _Outptr_ struct _ebpf_object** object,
+        _Out_ fd_t* program_fd,
+        _Outptr_result_maybenull_z_ const char** log_buffer);
+
+    /**
+     * @brief Get next program in ebpf_object object.
+     *
+     * @param[in] previous Pointer to previous eBPF program.
+     * @param[in] object Pointer to eBPF object.
+     * @return Pointer to the next program, or NULL if none.
+     */
+    _Ret_maybenull_ struct _ebpf_program*
+    ebpf_program_next(_In_opt_ const struct _ebpf_program* previous, _In_ const struct _ebpf_object* object);
+
+    /**
+     * @brief Get previous program in ebpf_object object.
+     *
+     * @param[in] next Pointer to next eBPF program.
+     * @param[in] object Pointer to eBPF object.
+     * @return Pointer to the previous program, or NULL if none.
+     */
+    _Ret_maybenull_ struct _ebpf_program*
+    ebpf_program_previous(_In_opt_ const struct _ebpf_program* next, _In_ const struct _ebpf_object* object);
+
+    /**
+     * @brief Get next map in ebpf_object object.
+     *
+     * @param[in] previous Pointer to previous eBPF map.
+     * @param[in] object Pointer to eBPF object.
+     * @return Pointer to the next map, or NULL if none.
+     */
+    _Ret_maybenull_ struct _ebpf_map*
+    ebpf_map_next(_In_opt_ const struct _ebpf_map* previous, _In_ const struct _ebpf_object* object);
+
+    /**
+     * @brief Get previous map in ebpf_object object.
+     *
+     * @param[in] next Pointer to next eBPF map.
+     * @param[in] object Pointer to eBPF object.
+     * @return Pointer to the previous map, or NULL if none.
+     */
+    _Ret_maybenull_ struct _ebpf_map*
+    ebpf_map_previous(_In_opt_ const struct _ebpf_map* next, _In_ const struct _ebpf_object* object);
+
+    /**
+     * @brief Fetch fd for a program object.
+     *
+     * @param[in] program Pointer to eBPF program.
+     * @return fd for the program on success, ebpf_fd_invalid on failure.
+     */
+    fd_t
+    ebpf_program_get_fd(_In_ const struct _ebpf_program* program);
+
+    /**
+     * @brief Fetch fd for a map object.
+     *
+     * @param[in] program Pointer to eBPF map.
+     * @return fd for the map on success, ebpf_fd_invalid on failure.
+     */
+    fd_t
+    ebpf_map_get_fd(_In_ const struct _ebpf_map* map);
+
+    /**
+     * @brief Clean up ebpf_object. Also delete all the sub objects
+     * (maps, programs) and close the related file descriptors.
+     *
+     * @param[in] obj Pointer to ebpf_object.
```

Comment:
```suggestion
     * @param[in] object Pointer to ebpf_object.
```
make param name match prototype